### DISCUSSION TITLE
Use an API external call to enrich the userinfo

### DISCUSSION THREAD

**Discussion by Amaury4sg** [OpenIDC/mod_auth_openidc#884] [Q&A] [answered]
Hello,

When I do a userinfo, I'd like to call an external API like http://myApi/?getByUserId=[userid]
add this to the cache of the attributes to be send in the headers each call.
And if the userinfo is to be refreshed for any reason, do another API call to http://myApi/?getByUserId=[userid]
I'd like to avoid doing the call on the OpenID Provider, but if there is no easy to do it with mod_auth_openidc that what we'll do.
I'm taking a two week vacation so no pressure (and apologies if I don't answer quickly)

thanks for your help

**Comment by zandbelt** [comment, marked as answer]
that's a non-standards compliant extension that will therefore not be implemented in this module, other than that, the userinfo endpoint is really meant to assemble user information to the Provider seems to be the right place to do so, instead of each RP

**Reply by Amaury4sg** [reply]
Understood, thanks for your time
on our side it's only going to be for a single RP, so we didn't want to add complexity to the OP
